src/serialize.js

```diff
@@ -38,6 +39,10 @@ function handleProperty(clone, key, value, valuePath, serializationSymbol, assig
         }
     } else if (value.constructor === Date) {
         assignments.push(valuePath + '=new Date(' + value.getTime() + ')');
+    } else if (value.constructor === URL) {
+        assignments.push(valuePath + '=new URL("' + value.href + '")');
+    } else if (value.constructor === URLSearchParams) {
+        assignments.push(valuePath + '=new URLSearchParams("' + value.toString() + '")');
```

Comment:
This should probably be escaped. Also I don't think the `toString` is necessary.